ee/codegen/src/__test__/graph-attribute.test.ts

```diff
@@ -686,5 +687,99 @@ describe("Workflow", () => {
         [startNode, endNode],
       ]);
     });
+
+    it("should be pointing to the correct terminal nodes from a nested set of conditionals", async () => {
+      const firstCheckNode = genericNodeFactory({
+        name: "FirstCheckNode",
+        nodePorts: [
+          nodePortFactory({
+            type: "IF",
+          }),
+          nodePortFactory({
+            type: "ELSE",
+          }),
+        ],
+      });
+
+      const firstInnerCheckNode = genericNodeFactory({
+        name: "FirstInnerCheckNode",
+        nodePorts: [
+          nodePortFactory({
+            type: "IF",
+          }),
+          nodePortFactory({
+            type: "ELSE",
+          }),
+        ],
+      });
+
+      const finalCheckNode = genericNodeFactory({
+        name: "FinalCheckNode",
+        nodePorts: [
+          nodePortFactory({
+            type: "IF",
+          }),
+          nodePortFactory({
+            type: "ELSE",
+          }),
+        ],
+      });
+
+      const innerTerminalNode = genericNodeFactory({
+        name: "InnerTerminalNode",
+      });
+
+      const secondInnerCheckNode = genericNodeFactory({
+        name: "SecondInnerCheckNode",
+        nodePorts: [
+          nodePortFactory({
+            type: "IF",
+          }),
+          nodePortFactory({
+            type: "ELSE",
+          }),
+        ],
+      });
+
+      const outerOutputNode = genericNodeFactory({
+        name: "OuterOutputNode",
+      });
+
+      await runGraphTest([
+        [entrypointNode, firstCheckNode],
+        [[firstCheckNode, "if_port"], firstInnerCheckNode],
+        [[firstCheckNode, "else_port"], finalCheckNode],
+        [[finalCheckNode, "if_port"], innerTerminalNode],
+        [[finalCheckNode, "else_port"], outerOutputNode],
+        [[secondInnerCheckNode, "if_port"], finalCheckNode],
+        [[secondInnerCheckNode, "else_port"], outerOutputNode],
+        [[firstInnerCheckNode, "if_port"], secondInnerCheckNode],
+        [[firstInnerCheckNode, "else_port"], finalCheckNode],
+      ]);
```

Comment:
This graph is why I had all those PRs cleaning up the graph attribute tests. I anticipate way more complex edge cases to come